frontend/src/utility_components/Toast.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect } from 'react';
+
+interface ToastProps {
+    message?: string; // Make the message prop optional
+    onClose: () => void;
+}
+
+const Toast: React.FC<ToastProps> = ({ message, onClose }) => {
+    useEffect(() => {
+        if (message) {
+            // Close the toast automatically after 5 seconds
+            const timer = setTimeout(onClose, 5000);
+
+            return () => clearTimeout(timer);
+        }
+    }, [message, onClose]);
+
+    if (!message) {
+        return null; // If no message, don't render the toast at all
+    }
+
+    return (
+        <div className="fixed top-0 left-0 w-full h-12 bg-blue-500 text-white text-center flex justify-center items-center">
```

Comment:
to handle whether if message exists, you can conditionally render the div like 
```
return (
(message && <div>whatever</div>)
)
```

(it might not be the right syntax cause im doing it from the top of my head)